src/backend/d3d12/BufferD3D12.cpp

```diff
@@ -153,14 +162,19 @@ namespace backend { namespace d3d12 {
     }
 
     void Buffer::SetSubDataImpl(uint32_t start, uint32_t count, const uint8_t* data) {
-        mDevice->GetResourceUploader()->BufferSubData(mResource, start, count, data);
+        Device* device = ToBackend(GetDevice());
+
+        TransitionUsageNow(device->GetPendingCommandList(), nxt::BufferUsageBit::TransferDst);
+        device->GetResourceUploader()->BufferSubData(mResource, start, count, data);
     }
 
     void Buffer::MapReadAsyncImpl(uint32_t serial, uint32_t start, uint32_t count) {
         D3D12_RANGE readRange = {start, start + count};
         char* data = nullptr;
         ASSERT_SUCCESS(mResource->Map(0, &readRange, reinterpret_cast<void**>(&data)));
 
+        // There is no need to transition the resource to a new state: D3D12 seems to make the GPU
```

Comment:
Not an entirely naive implementation, then. :)

Reply:
Yeah but wait until you rerun the skia-nxt benchmarks on Vulkan ^^